File: xxCommitter/Committer.cpp
#include "Committer.h"
#include "spdlog/spdlog.h"
#include <iostream>
#include <fstream>

Committer::Committer(const std::string &a_filepath)
	: m_filepath(a_filepath)
{
	spdlog::debug("Commits file is at {}", m_filepath);
}


bool Committer::CommitWithMsg(const std::string &a_msg) 
{
	// MISSING: Call "git commit -m msg" here, and any other git actions

	if (a_msg.empty())
	{
		return false;
	}

	std::ofstream out(m_filepath, std::ios_base::app);
	out << a_msg << "\n";
	return out.good();
}

void Committer::ShowPreviousCommits()
{
	std::ifstream file(m_filepath);
	if (!file.good()) {
        std::cout << "No previous commits found.";
		return;
	}

	while (file.good())
	{
		std::string line;
		std::getline(file, line);
		if (!line.empty())
		{
			std::cout << line << "\n";
		}
	}
}

File: sharpCommitter/input/Committer.cs
using Microsoft.Extensions.Logging;

namespace SharpCommitter
{
    class Committer
    {
        private static readonly string FILENAME = "prevcommits.txt"; // THIS SHOULD BE ON A SHARED NETWORK DRIVE/MOUNT and taken from configuration

        private readonly string m_filepath;

        public Committer(ILogger<Committer> a_logger)
        {
            m_filepath = FILENAME;
            a_logger.LogDebug("Commits file is at {filepath}", m_filepath);
        }

        public bool CommitWithMsg(string a_msg) 
        {
            // MISSING: Call "git commit -m msg" hereI and any other git actions

            File.AppendAllText(m_filepath, a_msg + "\n");
            return true;                
        }

        public void ShowPreviousCommits()
        {
            try
            {
                Console.WriteLine("Commit history:");
                string[] lines = File.ReadAllLines(m_filepath);
                foreach (string line in lines)
                {
                    Console.Write("* ");
                    Console.WriteLine(line);
                }
            }
            catch (FileNotFoundException)
            {
                Console.WriteLine("No previous commits found.");
            }
        }
    }
}
